### PR TITLE
Fix 404 home link on deeper 404 pages

### LINKED ISSUE #1437: 404 - click on home - when deeper than root
Won’t link to home but to another error page

### PR DESCRIPTION
**Pull request by RonaldDijkstra**
Force an absolute `full_url` for the link to home on any given 404 page, to make sure home links on 404's work everywhere.

Close #1437

### CODE REVIEW

**Comment by RonaldDijkstra** [comment]
```
- ./build/blog/medizinischer-fachangestellter-im-krankenhaus-entlastung-fur-den-arzt/index.html
  *  External link http://www.marburger-bund.de/ failed: 503 No error
```

**Comment by RonaldDijkstra** [comment]
Deployed to staging to test: 

Home with a relative URL
https://www.defacto.nl/404/404 

Home with an absolute URL
https://staging.defacto.nl/404/404

File: source/localizable/header/_main-nav.de.erb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <nav class="main-nav <%= header_style %>">
   <ul>
     <li>
-      <%= nav_link_to t("header.home"), "/" %>
+      <%= nav_link_to t("header.home"), nav_link_home %>
```
Comment (sn3p) [inline source/localizable/header/_main-nav.de.erb]: Maybe we should add parentheses when calling functions to distinguish them form variables:

```rb
nav_link_home()
```

File: config.rb [hunk 2]
```diff
@@ -486,13 +486,23 @@ def nav_link_to(*args, &block)
...
     options[:class] ||= ""
     options[:class] << " active" if is_active
 
     args[options_index] = options
     locale_link_to(*args, &block)
   end
 
+  # Force full_url on a homelink on any given 404 page
+  def nav_link_home
+    if current_page.url == "/404.html"
```
Comment (RonaldDijkstra) [inline config.rb]: #1434

File: source/localizable/partials/_header.erb [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
       <div class="logo">
 
-        <% locale_link_to("/") do %>
+        <% locale_link_to(nav_link_home) do %>
```
Comment (sn3p) [inline source/localizable/partials/_header.erb]: Add parentheses here as well?

File: source/localizable/header/_main-nav.de.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <nav class="main-nav <%= header_style %>">
   <ul>
     <li>
-      <%= nav_link_to t("header.home"), "/" %>
+      <%= nav_link_to t("header.home"), nav_link_home() %>
```
Comment (sn3p) [inline source/localizable/header/_main-nav.de.erb]:
```suggestion
      <%= nav_link_to t("header.home"), "/", relative: false %>
```

Works as well

Comment (RonaldDijkstra) [reply source/localizable/header/_main-nav.de.erb]: Nice 👍

**Comment by RonaldDijkstra** [comment]
Tested on staging, works

**Pull request merged**